src/test/scala/strawman/collection/test/Test.scala

```diff
@@ -288,6 +288,15 @@ class StrawmanTest {
     println(xs17.to(List))
   }
 
+  def mapOps(xs: strawman.collection.Map[Int, String]): Unit = {
+    // val xs1 = xs.map { case (k, v) => (v, k) }
+    // val xs2: strawman.collection.Map[String, Int] = xs1
+    val xs1 = xs.map(kv => (kv._2, kv._1))
+    val xs2: strawman.collection.Iterable[(String, Int)] = xs1
```

Comment:
You haven't given it any way to get `Map[String, Int]`, have you?  The standard trick is something like
```
def map[C, U](f: ((K, V)) => ((C, U)))(implicit ev: ((C, U)) <:< Tuple2[_, _]): Map[C, U] = ???
```
or somehow get implicit resolution in there.  That still gives broken type inference, but type inference and implicit resolution could be tweaked so that even *without* inferring all the types, the compiler would know to try this one first (and then be able to get through the case without an issue).  I unfortunately don't know whether the internals can support this with a reasonable amount of work, but conceptually it seems like it would solve the issue.